src/type/validate.ts

```diff
@@ -480,7 +480,20 @@ function validateUnionMembers(
   }
 
   const includedTypeNames = new Set<string>();
+  const includedInvalidTypes = new Set<string>();
   for (const memberType of memberTypes) {
+    if (!isObjectType(memberType)) {
+      const typeString = inspect(memberType);
+      if (!includedInvalidTypes.has(typeString)) {
+        context.reportError(
+          `Union type ${union.name} can only include Object types, ` +
+            `it cannot include ${inspect(memberType)}.`,
+          getUnionMemberTypeNodes(union, String(memberType)),
+        );
+        includedInvalidTypes.add(typeString);
+      }
+      continue;
```

Comment:
this reminds me of https://github.com/graphql/graphql-js/pull/4181 => what do you think about changing the error message string to something that references both errors, at least if there are repetitions?

'Something like Union type can only include unique Object types, ....'

What do you think?